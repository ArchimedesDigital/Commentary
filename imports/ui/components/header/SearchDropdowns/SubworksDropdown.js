import React from 'react';

// components:
import SearchToolDropdown from '/imports/ui/components/header/SearchToolDropdown'; // eslint-disable-line import/no-absolute-path
import SearchTermButton from '/imports/ui/components/header/SearchTermButton'; // eslint-disable-line import/no-absolute-path

// helpers:
import { isActive, dropdownPropTypes, dropdownDefaultProps } from './helpers';

/*
	BEGIN SubworksDropdown
*/
const SubworksDropdown = ({ works, searchDropdownOpen, toggleSearchDropdown, toggleSearchTerm, selectedWork, workInFilter, filters }) => (
	<SearchToolDropdown
		name={selectedWork}
		open={searchDropdownOpen === 'Book' || searchDropdownOpen === 'Hymn'}
		toggle={toggleSearchDropdown}
		disabled={!workInFilter}
	>
		{works.map((work) => {
			const workIsAvtive = isActive(filters, work, 'works');
			if (workIsAvtive) {
				const SearchTermButtons = work.subworks.map(subwork => (
					<SearchTermButton
						key={subwork.n}
						toggleSearchTerm={toggleSearchTerm}
						label={`${work.title} ${subwork.title}`}
						searchTermKey="subworks"
						value={subwork}
						active={isActive(filters, subwork, 'subworks', 'n')}
					/>
				));
				return SearchTermButtons;
			}
			return null;
		})}
	</SearchToolDropdown>
);
SubworksDropdown.propTypes = {
	works: React.PropTypes.arrayOf(React.PropTypes.shape({
		title: React.PropTypes.string.isRequired,
		subworks: React.PropTypes.arrayOf(React.PropTypes.shape({
			n: React.PropTypes.number.isRequired,
			title: React.PropTypes.string.isRequired,
		})),
	})),
	selectedWork: React.PropTypes.string.isRequired,
	workInFilter: React.PropTypes.bool.isRequired,
	...dropdownPropTypes,
};
SubworksDropdown.defaultProps = {
	works: [],
	...dropdownDefaultProps,
};
/*
	END SubworksDropdown
*/

export default SubworksDropdown;
