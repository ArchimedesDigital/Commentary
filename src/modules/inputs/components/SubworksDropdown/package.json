{
  "name": "SubworksDropdown",
  "version": "0.0.1",
  "private": true,
  "main": "./SubworksDropdown.js"
}
