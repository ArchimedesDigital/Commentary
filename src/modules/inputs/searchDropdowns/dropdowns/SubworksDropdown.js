import React from 'react';
import PropTypes from 'prop-types';

// components:
import SearchToolDropdown from '../../../../components/header/SearchToolDropdown';
import SearchTermButton from '../../../../components/header/SearchTermButton';

// helpers:
import { isActive, dropdownPropTypes, dropdownDefaultProps } from '../helpers';

/*
	BEGIN SubworksDropdown
*/
const SubworksDropdown = ({ works, searchDropdownOpen, toggleSearchDropdown, toggleSearchTerm, selectedWork, workInFilter, filters }) => (
	<SearchToolDropdown
		name={selectedWork}
		open={searchDropdownOpen === 'Book' || searchDropdownOpen === 'Hymn'}
		toggle={toggleSearchDropdown}
		disabled={!workInFilter}
	>
		{works.map((work) => {
			const workIsAvtive = isActive(filters, work, 'works');
			if (workIsAvtive) {
				const SearchTermButtons = work.subworks.map(subwork => (
					<SearchTermButton
						key={subwork.n}
						toggleSearchTerm={toggleSearchTerm}
						label={work.title === 'Homeric Hymns' ? `Hymn ${subwork.title}` : `${work.title} ${subwork.title}`}
						searchTermKey="subworks"
						value={subwork}
						active={isActive(filters, subwork, 'subworks', 'n')}
					/>
				));
				return SearchTermButtons;
			}
			return null;
		})}
	</SearchToolDropdown>
);
SubworksDropdown.propTypes = {
	works: PropTypes.arrayOf(PropTypes.shape({
		title: PropTypes.string.isRequired,
		subworks: PropTypes.arrayOf(PropTypes.shape({
			n: PropTypes.number.isRequired,
			title: PropTypes.string.isRequired,
		})),
	})),
	selectedWork: PropTypes.string.isRequired,
	workInFilter: PropTypes.bool.isRequired,
	...dropdownPropTypes,
};
SubworksDropdown.defaultProps = {
	works: [],
	...dropdownDefaultProps,
};
/*
	END SubworksDropdown
*/

export default SubworksDropdown;
