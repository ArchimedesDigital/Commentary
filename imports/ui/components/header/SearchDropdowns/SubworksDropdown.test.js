import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import SubworksDropdown from './SubworksDropdown';

describe('SubworksDropdown', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<SubworksDropdown
						selectedWork=""
						workInFilter={false}
						toggleSearchDropdown={() => {}}
						toggleSearchTerm={() => {}}
						toggle={() => {}}
					/>
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

