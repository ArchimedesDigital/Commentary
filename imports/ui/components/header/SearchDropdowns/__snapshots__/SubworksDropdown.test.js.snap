// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubworksDropdown renders correctly 1`] = `
<div
  className="dropdown search-dropdown search-dropdown-- "
>
  <button
    className="search-tool dropdown-toggle search-tool-disabled"
    disabled={true}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "rgba(0, 0, 0, 0)",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.3)",
        "cursor": "default",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "minWidth": 88,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transform": "translate(0, 0)",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="mdi mdi-chevron-down"
      color="rgba(0, 0, 0, 0.3)"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.3)",
          "display": "inline-block",
          "fontSize": 24,
          "marginLeft": 0,
          "marginRight": 0,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": "middle",
        }
      }
    />
  </button>
  <ul
    className="dropdown-menu"
  >
    <div
      className="dropdown-menu-inner"
    />
    <button
      className="close-dropdown"
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="mdi mdi-close"
          color="rgba(0, 0, 0, 0.87)"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        />
      </div>
    </button>
  </ul>
</div>
`;
